ntbs-integration-tests/NotificationPages/MBovisExposureToKnownCasesPageTests.cs

```diff
@@ -0,0 +1,184 @@
+
+        public MBovisExposureToKnownCasesPageTests(NtbsWebApplicationFactory<Startup> factory) : base(factory)
+        {
+        }
+        
+        public static IList<Notification> GetSeedingNotifications()
+        {
+            return new List<Notification>
+            {
+                new Notification
+                {
+                    NotificationId = Utilities.NOTIFICATION_ID_WITH_MBOVIS_AND_ENTITIES,
+                    NotificationStatus = NotificationStatus.Notified,
+                    DrugResistanceProfile = new DrugResistanceProfile {Species = "M. bovis"},
+                    MBovisDetails = new MBovisDetails
+                    {
+                        HasExposureToKnownCases = true,
+                        MBovisExposureToKnownCases = new List<MBovisExposureToKnownCase>
+                        {
+                            new MBovisExposureToKnownCase
+                            {
+                                ExposureSetting = ExposureSetting.Household,
+                                YearOfExposure = 2010,
+                                ExposureNotificationId = Utilities.NOTIFICATION_ID_WITH_MBOVIS_NO_ENTITIES
+                            }
+                        }
+                    }
+                },
+                new Notification
+                {
+                    NotificationId = Utilities.NOTIFICATION_ID_WITH_MBOVIS_NO_ENTITIES,
+                    NotificationStatus = NotificationStatus.Notified,
+                    DrugResistanceProfile = new DrugResistanceProfile {Species = "M. bovis"}
+                }
+            };
+        }
+
+        [Fact]
+        public async Task IfNotificationHasKnownCases_DisplaysTable()
+        {
+            // Arrange
+            const int notificationId = Utilities.NOTIFICATION_ID_WITH_MBOVIS_AND_ENTITIES;
+
+            // Act
+            var response = await Client.GetAsync(GetCurrentPathForId(notificationId));
+
+            // Assert
+            var document = await GetDocumentAsync(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.NotNull(document.QuerySelector("div[id='mbovis-exposure-to-known-cases-table']"));
+        }
+
+        [Fact]
+        public async Task IfNotificationDoesNotHaveKnownCases_DoesNotDisplayTable()
+        {
+            // Arrange
+            const int notificationId = Utilities.NOTIFICATION_ID_WITH_MBOVIS_NO_ENTITIES;
+
+            // Act
+            var response = await Client.GetAsync(GetCurrentPathForId(notificationId));
+
+            // Assert
+            var document = await GetDocumentAsync(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Null(document.QuerySelector("div[id='mbovis-exposure-to-known-cases-table']"));
+        }
+        
+        [Fact]
+        public async Task RedirectsToOverviewWithCorrectAnchorFragment()
```

Comment:
This could definitely be testable via unit testing, however it would need to be tested on all the view models (and for them to have all their interfaces mocked out - which would be a chunk of bloat for what can be achieved here quite simply - it is unfortunately duplicated across all the 1st tier edit pages...